metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowExecutor.java

```diff
@@ -148,13 +149,46 @@ private Date runMetisPlugin(AbstractMetisPlugin abstractMetisPlugin) {
     //Start execution and periodical check
     abstractMetisPlugin
         .execute(dpsClient, ecloudBaseUrl, ecloudProvider, workflowExecution.getEcloudDatasetId());
+
+    if (!abstractMetisPlugin.getPluginMetadata().isMocked()) {
+      return periodicCheckingLoop(sleepTime, abstractMetisPlugin);
+    } else {
+      return periodicCheckingLoopMocked(sleepTime, iterationsToFake, abstractMetisPlugin);
+    }
+  }
+
+  private Date periodicCheckingLoop(int sleepTime, AbstractMetisPlugin abstractMetisPlugin) {
```

Comment:
This method is very similar to periodicCheckingLoopMocked... is there no way to combine these two?

Reply:
I thought about that as well but the problem is the nature of the loops. I was going to capture in a separate method the code inside the loop but the problem is that the null values returned refer to Date and the taskState which is required for the external loop is TaskState so that is not possible. The other option is to complicate the loop in one method and take into consideration both scenarios, which I tried but I got complaints from Sonar of loop conditions complexity plus I felt it was less readable, so I decided readability over complexity. We can revise that if needed in the future.